### PR TITLE
Improve Documentation: Learn the Basics

### PR DESCRIPTION
**Pull request by Gistbatch**
### Description
I went through the "Learn the Basics" documentation and found some bugs in the examples.
I fixed them here.

### CODE REVIEW

**Comment by Gistbatch** [comment]
I have read the Contributor License Agreement and I hereby accept the Terms.

**Comment by amccaskey** [comment]
Thank you for these quick fixes!

**Review by amccaskey — APPROVED** [review]

File: docs/sphinx/using/cudaq/variational.rst [hunk 1]
```diff
@@ -94,7 +96,7 @@ Here is an example
         return cost;
       });
 
-CUDA Quantum packages the above up for the variational quantum eigensolver algorithm in 
+CUDA Quantum wraps the above code up for the variational quantum eigensolver algorithm in
```
Comment (schweitzpgi) [inline docs/sphinx/using/cudaq/variational.rst:99]: Maybe "re-exports" would be a more precise verb here?

**Review by schweitzpgi — APPROVED** [review]

File: docs/sphinx/using/cudaq/variational.rst (hunk 1 above)
Comment (Gistbatch) [reply docs/sphinx/using/cudaq/variational.rst:99]: I'm not a native English speaker, I just tripped over the formulation when reading it for the first time. I was reading "CUDA Quatnum packages" as the noun. So I'm open to any suggestions.

Comment (schweitzpgi) [reply docs/sphinx/using/cudaq/variational.rst:99]:
```suggestion
CUDA Quantum provides the above code for the variational quantum eigensolver algorithm in 
```
I think this minor tweak will do the trick.

File: docs/sphinx/using/cudaq/variational.rst [hunk 2]
```diff
@@ -10,6 +10,8 @@ signature for gradient-based optimizers, where the arguments represent the
 function parameters at the current iteration and the modifiable gradient
 vector reference, respectively. For gradient-free optimizers, the second 
 argument, :code:`double(const std::vector<double>&)`, can be dropped.
+Optimizers are defined in the header :code:`<cudaq/optimizers.h>` and 
+gradients in :code:`<cudaq/gradints.h>`.
```
Comment (amccaskey) [inline docs/sphinx/using/cudaq/variational.rst:14]: typo on `<cudaq/gradients.h>`

**Review by amccaskey — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**